thrust/thrust/random/detail/uniform_int_distribution.inl

```diff
@@ -75,7 +75,13 @@ uniform_int_distribution<IntType>::operator()(UniformRandomNumberGenerator& urng
   // XXX adding 1.0 to a potentially large floating point number seems like a bad idea
   uniform_real_distribution<float_type> real_dist(real_min, real_max + float_type(1));
 
-  return static_cast<result_type>(real_dist(urng));
+  // Get the base number from real distribution
+  result_type base = static_cast<result_type>(real_dist(urng));
+
+  // Add random lower bits to fix last 12-bits always being 0.
+  result_type lower_bits = static_cast<result_type>(urng()) & 0xFFF; // Get 12 **uniformly** random bits
+  return base | lower_bits; // Combine them
```

Comment:
Is there a particular reason we use the real_dist, instead of using an algorithm that can directly generate integers.

Also won't this break for int distributions < 12 bits

Reply:
> Is there a particular reason we use the real_dist, instead of using an algorithm that can directly generate integers.

It's legacy, unfortunately I don't know.

> Also won't this break for int distributions < 12 bits

yes, good catch. fixing it rn.